### ISSUE TITLE
Discord not running causes Rare to throw an exception

### ISSUE THREAD

**Issue by MultisampledNight** [Dummerle/Rare#41]
## **Describe the bug**
If Discord is not running while launching Rare, Rare crashes with this exception:
```py
Traceback (most recent call last):
  File "/home/multisn8/.local/lib/python3.9/site-packages/Rare-1.0.0-py3.9.egg/rare/components/main_window.py", line 34, in <lambda>
    self.tab_widget.settings.rare_settings.rpc.update_settings.connect(lambda: self.rpc.changed_settings(self.tab_widget.games_tab.default_widget.game_list.running_games))
  File "/home/multisn8/.local/lib/python3.9/site-packages/Rare-1.0.0-py3.9.egg/rare/utils/rpc.py", line 29, in changed_settings
    self.remove_rpc()
  File "/home/multisn8/.local/lib/python3.9/site-packages/Rare-1.0.0-py3.9.egg/rare/utils/rpc.py", line 43, in remove_rpc
    self.set_discord_rpc()
  File "/home/multisn8/.local/lib/python3.9/site-packages/Rare-1.0.0-py3.9.egg/rare/utils/rpc.py", line 48, in set_discord_rpc
    self.RPC.connect()
  File "/home/multisn8/.local/lib/python3.9/site-packages/pypresence/presence.py", line 43, in connect
    self.loop.run_until_complete(self.handshake())
  File "/usr/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/home/multisn8/.local/lib/python3.9/site-packages/pypresence/baseclient.py", line 128, in handshake
    self.sock_reader, self.sock_writer = await asyncio.open_unix_connection(self.ipc_path, loop=self.loop)
  File "/usr/lib/python3.9/asyncio/streams.py", line 111, in open_unix_connection
    transport, _ = await loop.create_unix_connection(
  File "/usr/lib/python3.9/asyncio/unix_events.py", line 244, in create_unix_connection
    await self.sock_connect(sock, path)
  File "/usr/lib/python3.9/asyncio/selector_events.py", line 502, in sock_connect
    return await fut
  File "/usr/lib/python3.9/asyncio/selector_events.py", line 507, in _sock_connect
    sock.connect(address)
ConnectionRefusedError: [Errno 111] Connection refused
zsh: abort (core dumped)  LANG=en_US.UTF-8 rare
```

## **To Reproduce**
Steps to reproduce the behavior depend on if you have the option "Discord RPC only when playing" enabled.
If you have it enabled:
1. Make sure Discord is closed
2. Start up Rare in your terminal emulator
3. Launch any game
4. See Rare throwing the exception in the terminal emulator
Or:
3. Go to the settings in the upper right corner
4. Click on the dropdown below `Discord RPC` and change to `Show always`
5. See Rare throwing the exception in the terminal emulator

If you have the setting on `Show always`, starting up Rare is enough to see the exception.

The bug does not appear if the setting is set to `Show never`.

## **Expected behavior**
That Rare continues running normally.

## **Screenshots**
![lol](https://user-images.githubusercontent.com/80128916/114532382-15f93300-9c4d-11eb-9ac6-71fc9228b102.png)

## **System information**
 - OS: Arch Linux, rolling release and fully updated
 - Python version: Python 3.9.2
 - Last Rare commit: 5c6ae2b761c6920b4d519aabc62f64adf78ce774

## **Additional context**
It's probably just about a `try` and an `except` to catch the `ConnectionRefusedError`.

Is it better if I report bugs here on GitHub or on Discord? Getting stuff from the EGL run has never been easier. Thanks! :)

**Comment by Dummerle** [comment]
This should be fixed now. You can test is

**Comment by MultisampledNight** [comment]
Yes, it seems to be fixed by 0acc9c5bdaff93b6caa7481c2e76986cbfb1e2d6. Thanks! Closing.
